app/models/manageiq/providers/openshift/container_manager/refresher.rb

```diff
@@ -3,9 +3,24 @@ module Openshift
     class ContainerManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
       include ::EmsRefresh::Refreshers::EmsRefresherMixin
 
+      def self.entities
+        %w(routes projects)
+      end
+
       def parse_inventory(ems, _targets = nil)
-        all_entities = ems.with_provider_connection(&:all_entities)
-        all_entities.merge!(ems.with_provider_connection(:service => ManageIQ::Providers::Kubernetes::ContainerManager.ems_type, &:all_entities))
+        all_entities = ems.with_provider_connection { |client| fetch_entities(client, self.class.entities) }
+        fetch = proc do |client|
+          fetch_entities(
+            client,
+            ManageIQ::Providers::Kubernetes::ContainerManager::Refresher.entities
+          )
+        end
+        all_entities.merge!(
+          ems.with_provider_connection(
+            :service => ManageIQ::Providers::Kubernetes::ContainerManager.ems_type,
+            &fetch
```

Comment:
@simon3z we talked about inlining fetch from above. I could not do so:
The only syntax that works for me to define a proc inside a function call is the {} syntax (the Proc.new do |x| end does not work) and rubocop does not like {} definitions that are more then 1 line long.
@simon3z @blomquisg any ideas?